docs/topics/db/models.txt

```diff
@@ -844,6 +844,26 @@ built-in model methods, adding new arguments. If you use ``*args,
 **kwargs`` in your method definitions, you are guaranteed that your
 code will automatically support those arguments when they are added.
 
+If you wish to update a field in the :meth:`~Model.save` method, you may also
+want to have this field automatically added to the ``update_fields`` kwarg.
+See :doc:`specifying which fields to save </ref/models/instances>`.
+This will ensure the field is saved when ``update_fields`` is specified.
+For example::
+
+
+    from django.db import models
+    from django.utils.text import slugify
+
+    class Blog(models.Model):
+        name = models.CharField(max_length=100)
+        slug = models.TextField()
+
+        def save(self, *args, **kwargs):
+            self.slug = slugify(self.name)
+            if update_fields is not None and "name" in update_fields:
+                update_fields = set(update_fields) | {"slug"}
```

Comment:
`update_fields` is a `NameError` here I think? 
You'd either need to supply the full signature in `def save(self, ..., update_Ffields...):` _or_ pluck it out of kwargs (`if kwargs and 'update_fields' in kwargs and kwargs['update_fields']` or whatever)

Reply:
Yes! :facepalm: Thank you! I've updated

Reply:
If `update_fields` is not a already a `set` we need two lots of set construction here. Given that you're forced to create one it might be better to structure this like so to avoid iterating  over `update_fields` twice:
```suggestion
            if update_fields is not None and "name" in update_fields:
                update_fields = {"slug"}.union(update_fields)
```

Reply:
Updated :+1: